core/src/main/java/org/apache/brooklyn/core/location/access/BrooklynAccessUtils.java

```diff
@@ -59,46 +60,52 @@
 
     public static HostAndPort getBrooklynAccessibleAddress(Entity entity, int port) {
         String host;
-        
+
+        EntityInternal entityInternal = (EntityInternal) entity;
+
         // look up port forwarding
         PortForwardManager pfw = entity.getConfig(PORT_FORWARDING_MANAGER);
-        if (pfw!=null) {
-            Collection<Location> ll = entity.getLocations();
-            
-            synchronized (BrooklynAccessUtils.class) {
-                // TODO finer-grained synchronization
-                
-                for (MachineLocation machine : Iterables.filter(ll, MachineLocation.class)) {
-                    HostAndPort hp = pfw.lookup(machine, port);
+        if (pfw == null) {
+            log.debug("No PortForwardManager, using default");
+            pfw = (PortForwardManager) entityInternal.getManagementContext().getLocationRegistry().getLocationManaged("portForwardManager(scope=global)");
+        }
+
+        Collection<Location> ll = entity.getLocations();
+
+        synchronized (BrooklynAccessUtils.class) {
```

Comment:
Do we need to check for null `pfw` here? Is it always there? What about non-jclouds locations?